packages/next/next-server/server/render.tsx

```diff
@@ -959,16 +962,31 @@ export async function renderToHTML(
     }
   }
 
-  html = await postProcess(
-    html,
-    {
-      getFontDefinition,
-    },
-    {
-      optimizeFonts: renderOpts.optimizeFonts,
-      optimizeImages: renderOpts.optimizeImages,
-    }
-  )
+  if (renderOpts.optimizeFonts || renderOpts.optimizeImages) {
+    html = await postProcess(
+      html,
+      {
+        getFontDefinition,
+      },
+      {
+        optimizeFonts: renderOpts.optimizeFonts,
+        optimizeImages: renderOpts.optimizeImages,
+      }
+    )
+  }
+
+  if (renderOpts.optimizeCss) {
+    const cssOptimizer = new Critters({
```

Comment:
is there no way we can share the parsed html content to critters?

Reply:
Yes, good point. I will add that as a next step to integrate the HTML parsing component